test/topology_custom/test_mv_backlog.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+import requests
+import re
+
+logger = logging.getLogger(__name__)
+
+async def wait_for_views(cql, mvs_count, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        logger.info(f"Views built: {done[0][0]}")
+        if done[0][0] == node_count * mvs_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+def get_metrics(addr):
+    # The Prometheus API is on port 9180, and always http
+    addr = 'http://' + addr + ':9180/metrics'
+    resp = requests.get(addr)
+    if resp.status_code != 200:
+        pytest.skip('Metrics port 9180 is not available')
+    response = requests.get(addr)
+    assert response.status_code == 200
+    return response.text
+
+def get_metric(metrics, name, requested_labels=None):
+    the_metrics = get_metrics(metrics)
+    total = 0.0
+    lines = re.compile('^'+name+'{.*$', re.MULTILINE)
+    for match in re.findall(lines, the_metrics):
+        a = match.split()
+        metric = a[0]
+        val = float(a[1])
+        # Check if match also matches the requested labels
+        if requested_labels:
+            # we know metric begins with name{ and ends with } - the labels
+            # are what we have between those
+            got_labels = metric[len(name)+1:-1].split(',')
+            # Check that every one of the requested labels is in got_labels:
+            for k, v in requested_labels.items():
+                if not f'{k}="{v}"' in got_labels:
+                    # No match for requested label, skip this metric (python
+                    # doesn't have "continue 2" so let's just set val to 0...
+                    val = 0
+                    break
+        total += float(val)
+    return total
+
+def get_replicas(cql, key):
+    return cql.cluster.metadata.get_replicas("ks", Int32Type.serialize(key, cql.cluster.protocol_version))
+
+# This test reproduces issue #18542
+# In the test, we create a table and perform a write to it a couple of times
+# Each time, we check that a view update backlog on some shard increased
+# due to the write.
```

Comment:
I doubt I can explain this better as a response to a comment than what I wrote in the issue but hopefully by answering questions I can provide the missing insight

> Second, can you say a few words (here in the discussion or in the commit message) on how this test failed before this PR? What actually happened? Was the view update backlog seen by this test zero? Why?

Right, the backlog seen by the test was zero. This happened because of the following:
0) all nodes started with empty backlogs
1) first node got the write request and became the coordinator of the request
2) the coordinator sent a replica write request to the second node
3) the second node received the replica write request on a shard different than the shard that contained the data that needed to be modified. (When I was referring to the `coordinator shard` in the description I meant that shard which received the write request on the right replica.)
4) the second node (on coordinator shard) invoked the correct shard to apply the write
5) the correct shard applied the write, generating view updates and consuming units from the `view_update_concurrency_semaphore` (even though the view update backlog is based on this semaphore, this is **not** increasing the view update backlog propagated in responses and gossip.)
6) the coordinator shard on the second node finished the invocation on the shard with updated data and then called `storage_proxy::get_view_update_backlog` using its result for the write response. 
7) the coordinator node received the response and updates its saved max backlog for the second node

The problem is that on the shard whose backlog in fact increased, we never updated the backlog used for the responses and gossip.
 
There can be a few sources of confusion here:
* for each shard on each node there are actually 2 view update backlogs:
    1. https://github.com/scylladb/scylladb/blob/2d9142250eac278286428f52f62d339691d3598a/replica/database.hh#L1847-L1849
    2. https://github.com/scylladb/scylladb/blob/2d9142250eac278286428f52f62d339691d3598a/db/view/node_view_update_backlog.hh#L36

The i. backlog is always up-to-date (it's the source of truth). The ii. backlog needs to be updated, but it's the backlog which we're using to select a max backlog to use in a response or in gossip.
* the ii. backlog is updated using the backlog getter:
https://github.com/scylladb/scylladb/blob/2d9142250eac278286428f52f62d339691d3598a/service/storage_proxy.cc#L2449-L2451
https://github.com/scylladb/scylladb/blob/2d9142250eac278286428f52f62d339691d3598a/db/view/view.cc#L2612-L2613

As a result, if we never "get" the view update backlog on the shard whose i. backlog increased, the ii. backlog for that shard will never increase, and other nodes will never get the information about a potential overload.